controllers/topology/cluster_controller_test.go

```diff
@@ -0,0 +1,377 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilfeature "k8s.io/component-base/featuregate/testing"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/contract"
+	"sigs.k8s.io/cluster-api/feature"
+	"sigs.k8s.io/cluster-api/internal/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestClusterReconciler_reconcile(t *testing.T) {
+	g := NewWithT(t)
+	timeout := 5 * time.Second
+
+	ns, err := env.CreateNamespace(ctx, "test-topology-cluster-reconcile")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	clusterName := "cluster1"
+	clusterClassName := "class1"
+	workerClassName := "linux-worker"
+	workerClassName2 := "windows-worker"
+
+	// The below objects are created in order to feed the reconcile loop all the information it needs to create a full
+	// Cluster given a skeletal Cluster object and a ClusterClass. The objects include:
+
+	// 1) Templates for Machine, Cluster, ControlPlane and Bootstrap.
+	infrastructureMachineTemplate := builder.InfrastructureMachineTemplate(ns.Name, "inframachinetemplate").Build()
+	infrastructureClusterTemplate := builder.InfrastructureClusterTemplate(ns.Name, "infraclustertemplate").
+		// Create spec fake setting to assert that template spec is non-empty for tests.
+		WithSpecFields(map[string]interface{}{"spec.template.spec.fakeSetting": true}).
+		Build()
+	controlPlaneTemplate := builder.ControlPlaneTemplate(ns.Name, "cp1").
+		WithInfrastructureMachineTemplate(infrastructureMachineTemplate).
+		Build()
+	bootstrapTemplate := builder.BootstrapTemplate(ns.Name, "bootstraptemplate").Build()
+
+	// 2) ClusterClass definitions including definitions of MachineDeploymentClasses used inside the ClusterClass.
+	machineDeploymentClass := builder.MachineDeploymentClass(ns.Name, "md1").
```

Comment:
```suggestion
	machineDeploymentClass1 := builder.MachineDeploymentClass(ns.Name, "md1").
```
minor nit: If there is another one with the 2 suffix I think it reads better to have a 1 suffix here. (same for machineDeploymentTopology2)